activesupport/lib/active_support/descendants_tracker.rb

```diff
@@ -17,8 +17,44 @@ def direct_descendants(klass)
       end
     end
 
-    if RubyFeatures::CLASS_DESCENDANTS
+    @clear_disabled = false
+
+    if RubyFeatures::CLASS_SUBCLASSES
+      @@excluded_descendants = if RUBY_ENGINE == "ruby"
+        # On MRI `ObjectSpace::WeakMap` keys are weak references.
+        # So we can simply use WeakMap as a `Set`.
+        ObjectSpace::WeakMap.new
+      else
+        # On TruffleRuby `ObjectSpace::WeakMap` keys are strong references.
+        # So we use `object_id` as a key and the actual object as a value.
+        #
+        # JRuby for now doesn't have Class#descendant, but when it will, it will likely
+        # have the same WeakMap semantic than Truffle so we future proof this as much as possible.
+        class WeakSet
+          def initialize
+            @map = ObjectSpace::WeakMap.new
+          end
+
+          def [](object)
+            @map.key?(object.object_id)
+          end
+
+          def []=(object, _present)
+            @map[object_id] = object
+          end
+        end
```

Comment:
I think this may need a `WeakSet.new` after this line.  Or am I misreading?

Reply:
No you are right I'm an idiot -_-

Reply:
https://github.com/rails/rails/commit/ff5e909b635bdc6ec0f915e101a44281dfd30b97